zcash_client_sqlite/src/scan.rs

```diff
@@ -372,7 +374,16 @@ pub fn decrypt_and_store_transaction<P: AsRef<Path>>(
         .collect::<Result<Result<Option<_>, _>, _>>()??
         .ok_or(Error(ErrorKind::IncorrectHRPExtFVK))?;
 
-    let outputs = decrypt_transaction(tx, &extfvks);
+    // Height is block height for mined transactions, and the "mempool height" (chain height + 1) for mempool transactions.
+    let last_height = data.query_row("SELECT MAX(height) FROM blocks", NO_PARAMS, |row| {
+        row.get(0).or(Ok(SAPLING_ACTIVATION_HEIGHT - 1))
+    })?;
+    let mut stmt_select_block = data.prepare("SELECT block FROM transactions WHERE txid = ?")?;
+    let height = stmt_select_block.query_row(&[tx.txid().0.to_vec()], |row| {
+        row.get(0).or(Ok(last_height + 1))
+    })?;
```

Comment:
Nit: `last_height` is queried even when it isn't needed.

Reply:
moved the `last_height` query into a code block inside `row.get(0).or({let last_height = ...})`